File: src/CMDemo.API/Models/Product.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace CMDemo.API.Models
{
    public class Product
    {
        [BsonId]
        public ObjectId _Id { get; set; }

        [BsonElement("id")]
        public int Id { get; set; }

        [BsonElement("sku")]
        public string SKU { get; set; }

        [BsonElement("name")]
        public string Name { get; set; }

        [BsonElement("price")]
        public double Price { get; set; }

        [BsonElement("attribute")]
        public Attribute Attribute { get; set; }
    }
}

File: src/CMDemo-Ng2-Cli/src/app/products/product.ts

export interface Product {
    id: number;
    sku: string;
    name: string;
    price: number;
    attribute: {
        fantastic: {
            value: boolean;
            type: number;
            name: string;
        },
        rating: {
            name: string;
            type: number;
            value: number;
        }
    };
}
